tests/basics/BuiltinsTest.py

```diff
@@ -524,3 +524,39 @@ def __complex__(self):
     print("Oops, must not happen.")
 else:
     print("Oops, must not happen.")
+
+# Test for any builtins
+print(any([None, None, None]))
+print(any([None, 4, None]))
+print(any([]))
+print(any(range(20)))
```

Comment:
Because it is so small, 20 elements only, it is optimized into a list soon. Only ranges larger than 256 elements are not optimized, so I suggest you go with 260 elements, and try all kinds of ranges, with one, two and 3 arguments, each time making sure it's more than 256 elements.

Reply:
Okay. So, I have added tests with >256 elements. But I'm failing those tests because of this assert statement https://github.com/Nuitka/Nuitka/pull/246/files/ba4ad49960f2fcac76973cf6c503210e3bc7a6af#diff-bcd50821425789a3c9bfd17b9c3d16f1R229

I tried to move them into a try-catch block but that didn't help either.

Reply:
So, Instead of asserting should I return `None`? Is that a good idea?

Reply:
Your assert looks suspicious, can it be that you are comparing a function and an integer? That is going to give pretty random results.

Reply:
Also, why would be size matter. We took care in the `any` node to not try too many things, by aborting after 256 checks, that could be good enough, but it means the user needs to take care, but the producer of the handle, it cannot tell if that is an issue to you or not, so I think it should just represent the best possible knowledge.

Reply:
And another thing I didn't check immediately, but are you making sure that e.g. `getLow().getIntegerValue()` is actually working and not returning `None`. I didn't yet see that, but it would definitely be needed to have `None` tests.

Also generally, I am in favor of delaying as much work as possible to until when it's needed. I was going to discuss the responsibilities of these classes with you online though. But generally, I think the `__init__` ought to take note only of what is absolutely necessary and no more. When too much is done in the constructor, it can quickly become too much overhead for what is probably a simple question.

But I also think we need to draw a line, and accept this at some point when it's working, and then cleanups come later. But correctness is not there yet fully.

Reply:
> Your assert looks suspicious, can it be that you are comparing a function and an integer? That is going to give pretty random results.

I think Assert is fine as it is giving AssertionError which is because the number of elements is greater than 256

Reply:
> And another thing I didn't check immediately, but are you making sure that e.g. getLow().getIntegerValue() is actually working and not returning None. I didn't yet see that, but it would definitely be needed to have None tests.

As the get methods are in the Base class, I have put them in try catch block which will catch if `getHigh().getIntergerValue()` gives an attribute error which it will when `getHigh()` returns a `None`.  https://github.com/Nuitka/Nuitka/pull/246/files#diff-bcd50821425789a3c9bfd17b9c3d16f1R223

Reply:
I really, really dislike that kind of error handling. I prefer them to be checks, please change these things to explicit checks. 

E.g. will `getHigh().getIntergerValue()` hopefully always raise an `AttributeError` and that is very hard to spot (misspelling), and a failed re-factoring, etc. could also cause this. Code should never be that fragile. I could well imagine something deep behind that function to raise an `AttributeError` due to other programming mistakes.

Reply:
And this is definitely wrong in one way or the other: `assert constant_value.getIterationLength < 10000`, either it is a coding rule violation, namely an integer attribute named like a function, or a function not being called.

I think you do this here:

```
>>> class C(object):
...    def f():
...       pass
... 
>>> C().f < 10000
False
```

The value is pretty random. I believe, it is comparing the pointer to the object that `False` is to the bound function object, and that can go either way.

Since this is a common mistake, in Python3, it gives: `TypeError: unorderable types: method() < int()` but not so in Python2.

Reply:
Just to be clear, avoid using exception handling for control flow at all costs. When an exception hits, your first instinct seems to be to catch it. Instead, you need to make sure it doesn't raise.

The first approach leads to very fragile and potentially slow code that makes it hard to understand the context you are in. If things can happen in an exception handler, and mean `AttributeError` is supposed to me e.g. `something is None` that is very un-obvious.

Reply:
> And this is definitely wrong in one way or the other: `assert constant_value.getIterationLength < 10000`, either it is a coding rule violation, namely an integer attribute named like a function, or a function not being called.
> 
> I think you do this here:
> 
> ```
> >>> class C(object):
> ...    def f():
> ...       pass
> ... 
> >>> C().f < 10000
> False
> ```
> 
> The value is pretty random. I believe, it is comparing the pointer to the object that `False` is to the bound function object, and that can go either way.
> 
> Since this is a common mistake, in Python3, it gives: `TypeError: unorderable types: method() < int()` but not so in Python2.

Yes, I know that I was missing `()` after `getIterationLength` and I have already updated it. You are probably looking at an older commit.